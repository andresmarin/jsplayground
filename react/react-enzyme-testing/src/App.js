import React from 'react';
import logo from './logo.svg';
import './App.css';

export default class App extends React.Component {

  state = {
    intro: null
  }

  changeIntro = ()=>{
    this.setState({ intro: "Good! You preseed the silly button." })
  }

  render() {

    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">{this.props.msg}</h1>
        </header>
        <p className="App-intro">
          {this.state.intro !== null ? this.state.intro : "This is just a silly test."}
  </p>
        <button type="button" onClick={this.changeIntro}>
          A silly Button
    </button>
      </div>
    )
  }

}


