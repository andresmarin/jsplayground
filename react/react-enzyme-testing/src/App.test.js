
import React from 'react';
import { shallow, configure } from 'enzyme';
import App from './App';
import Adapter from 'enzyme-adapter-react-16';


configure({ adapter: new Adapter() });



describe('App', () => {

  const tree = shallow(
    <App msg="It works!" />
  );

  it('should be defined', () => {
    expect(App).toBeDefined();
  });


  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('should render the app', () => {
    expect(tree.find('.App')).toBeTruthy();
  });


  it('contains 1 div with class App', () => {
    expect(tree.find('.App').length).toBe(1);
  });


  it('should have a welcome message', () => {
    expect(tree.find('.App-title').text()).toEqual("It works!");
  });

  it('should render a button with some text on it', () => {
    expect(tree.find('button').length).toBe(1);
    expect(tree.find('button').text()).toEqual("A silly Button");
  });

  it("Sould change the intro when the button is clicked", () => {
    expect(tree.find('.App-intro').length).toBe(1);

    //Before clicking the button
    expect(tree.find('.App-intro').text()).toEqual("This is just a silly test.");

    //Button click
    tree.find('button').simulate('click');
    //expect(changeIntro.calledOnce).to.equal(true);

    //After clicking the button
    expect(tree.find('.App-intro').text()).toEqual("Good! You preseed the silly button.");

  });

});
