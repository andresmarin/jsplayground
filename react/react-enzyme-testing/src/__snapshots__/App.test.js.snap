// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    msg="It works!"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
>
          <img
                    alt="logo"
                    className="App-logo"
                    src="logo.svg"
          />
          <h1
                    className="App-title"
          >
                    It works!
          </h1>
</header>,
        <p
          className="App-intro"
>
          This is just a silly test.
</p>,
        <button
          onClick={[Function]}
          type="button"
>
          A silly Button
</button>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
/>,
            <h1
              className="App-title"
>
              It works!
</h1>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "It works!",
              "className": "App-title",
            },
            "ref": null,
            "rendered": "It works!",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is just a silly test.",
          "className": "App-intro",
        },
        "ref": null,
        "rendered": "This is just a silly test.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "A silly Button",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "A silly Button",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
>
            <img
                        alt="logo"
                        className="App-logo"
                        src="logo.svg"
            />
            <h1
                        className="App-title"
            >
                        It works!
            </h1>
</header>,
          <p
            className="App-intro"
>
            This is just a silly test.
</p>,
          <button
            onClick={[Function]}
            type="button"
>
            A silly Button
</button>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
/>,
              <h1
                className="App-title"
>
                It works!
</h1>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "It works!",
                "className": "App-title",
              },
              "ref": null,
              "rendered": "It works!",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is just a silly test.",
            "className": "App-intro",
          },
          "ref": null,
          "rendered": "This is just a silly test.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "A silly Button",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "A silly Button",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
